src/format/write.rs

```diff
@@ -81,10 +81,33 @@ impl Write for &mut [u8] {
 #[cfg(feature = "alloc")]
 #[cfg_attr(docsrs, doc(cfg(feature = "alloc")))]
 impl Write for Vec<u8> {
-    fn write(&mut self, buf: &[u8]) -> Result<usize, Error> {
-        self.try_reserve(buf.len())?;
-        self.extend_from_slice(buf);
-        Ok(buf.len())
+    fn write(&mut self, data: &[u8]) -> Result<usize, Error> {
+        self.try_reserve(data.len())?;
+        self.extend_from_slice(data);
+        Ok(data.len())
+    }
+}
+
+/// Wrapper for a [`std::io::Write`] writer.
+#[cfg(feature = "std")]
+#[cfg_attr(docsrs, doc(cfg(feature = "std")))]
+pub(crate) struct IoWrite<'a> {
+    /// Inner writer.
+    inner: &'a mut dyn std::io::Write,
```

Comment:
this is a neat way of doing this!

Reply:
I did the same thing for `core::fmt::Write` in https://github.com/artichoke/strftime-ruby/pull/50. Let me know what you think!